drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImplTest.java

```diff
@@ -445,6 +448,46 @@ public void testPinnedModeHidden() throws Exception {
         verify(pinnedModeIndicator).setVisible(false);
     }
 
+    @Test
+    public void testRemoveDecisionTableWhenPinned() {
+        final Command callback = mock(Command.class);
+        final GridPinnedModeManager.PinnedContext context = mock(GridPinnedModeManager.PinnedContext.class);
+
+        when(defaultGridLayer.isGridPinned()).thenReturn(true);
+        when(defaultGridLayer.getPinnedContext()).thenReturn(context);
+        when(context.getGridWidget()).thenReturn(dtableView);
+
+        view.removeDecisionTable(dtableView,
+                                 callback);
+
+        verify(defaultGridLayer,
+               times(1)).exitPinnedMode(commandArgumentCaptor.capture());
+
+        final Command command = commandArgumentCaptor.getValue();
+        assertNotNull(command);
+        command.execute();
+
+        verify(defaultGridLayer,
+               times(1)).remove(dtableView);
+        verify(callback,
+               times(1)).execute();
+        verify(view,
+               times(1)).disableButtonMenu();
+        verify(defaultGridLayer,
+               times(1)).batch();
+    }
+
+    @Test
+    public void testRemoveDecisionTableWhenNotPinned() {
+        final Command callback = mock(Command.class);
+
+        view.removeDecisionTable(dtableView,
+                                 callback);
+
+        verify(callback,
+               times(1)).execute();
```

Comment:
Comes to my mind, not sure if possible, If the scope of the test enables (enough mocks present), check menu disabled - because of removing table, and then check menu is enabled because of another available table. Let me know if it makes sense, even more, if it is possible, otherwise keep as it is.

Reply:
This [test](https://github.com/kiegroup/drools-wb/blob/master/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImplTest.java#L321) checks the button is enabled when a table is selected.

And this [test](https://github.com/kiegroup/drools-wb/blob/master/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerPresenterTest.java#L467) (and it's "not pinned" friend) check the method tested above is invoked when the CDI event is received selecting a table.

Is that sufficient?

Reply:
Looking on the current tesst and the issue, I believe yes.